### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by tkaufman**
@searls I can remove the API stuff if you think it's just baggage from branching off your branch, let me know.

Otherwise, I think these changes get us a denormalized entry table and a simple view for seeing entries on an invoicing basis. Next up, charts!

### CODE REVIEW

File: app/models/dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    @days_not_billed_by_project = days_not_billed_by_project
+  end
+
+  def self.gather(week)
+    weeks = [week.previous, week]
+    agents = User.all.select {|a| a.billable_days_within(week.previous.beginning, week.end) > 0}
+    days_available = agents.collect {|a| a.billable_days_within(week.previous.beginning, week.end)}.reduce(:+)
+    all_entries = DenormalizedEntry.between_inclusive(weeks[0].beginning, weeks[1].end) || []
+    days_billed = all_entries.select{|e| e.billable}.collect{|e| e.amount}.reduce(:+)
+    days_not_billed = all_entries.select{|e| !e.billable}.collect{|e| e.amount}.reduce(:+)
```
Comment (searls) [inline app/models/dashboard.rb]: This could be a little terser with `all_entries.reject(&:billable).map(&:amount).sum`? (I think -- didn't test)

Comment (searls) [inline app/models/dashboard.rb]: If someone erroneously labels a day as full-day client project + full-day internal, will this count it as both 1 day billed & not billed?

**Comment by searls** [comment]
Seems pretty legit! Coding is fun!

Only comments:
- [ ] feel free to whack my API controller since it's probably dead weight
- [ ] Along the same lines, Does this PR rely on your fake data rake test? Should that be excluded in favor of fixtures? Interested to hear how that worked out for you

File: app/models/dashboard.rb (hunk 1 above)
Comment (tkaufman) [inline app/models/dashboard.rb]: No, those would be two separate entries which would get separated into billable and non billable groupings. 

I left the select in lieu of the reject to keep symmetry between days_billed and days_not_billed

**Comment by tkaufman** [comment]
I nuked the api stuff and the fake data rake tasks. My beef with fixtures is that it separates a lot of the relevant data into a different file so the tests become a leap of faith. That and I created a bunch of hard to find bugs because enums in fixtures are fun!

**Pull request merged**